src/client.rs

```diff
@@ -173,6 +173,22 @@ impl ImageLayer {
     }
 }
 
+/// Stream response of a blob with optional content length if available
+pub struct SizedStream {
+    /// The length of the stream if the upstream registry sent a `Content-Length` header
+    pub content_length: Option<u64>,
+    /// The stream of bytes
+    pub stream: BoxStream<'static, std::result::Result<bytes::Bytes, std::io::Error>>,
+}
+
+/// The response of a partial blob request
+pub enum PartialBlobResponse {
```

Comment:
I think this should probably just be called `BlobResponse` because it reads a little weird to have `PartialBlobResponse::Partial` (repetitive) and `PartialBlobResponse::Full` (contradictory). I know it is technically only being returned from the blob partial method, so I could definitely be argued out of it.

Reply:
Yeah, I was unsure about that. Initially I had it named `BlobResponse`, but then it felt weird, because with that name you would kinda expect that the `pull_blob_stream(...)` function would always return a `BlobResponse::Full`, which wouldn't make much sense to always wrap that response if `Full` is the only variant you get from that method? :man_shrugging: So I ended up renaming it. But you are right, the `PartialBlobResponse` name also is weird.

But since I was unsure which name is better anyway, I'm also happy to just rename it back. :+1: